packages/react/src/utils/custom-element.ts

```diff
@@ -26,4 +26,7 @@ export const createComponent = <I extends HTMLElement, E extends EventNames = {}
     },
   )
 
+  return output as React.ComponentType<React.ComponentProps<typeof output> & {sx?: unknown}>
```

Comment:
should `sx` be the standard `BetterSystemStyleObject` type here? From `SxProp`?:

```suggestion
  return output as React.ComponentType<React.ComponentProps<typeof output> & SxProp>
```